src/main/java/com/google/codeu/data/Message.java

```diff
@@ -42,6 +42,18 @@ public Message(UUID id, String user, String text, long timestamp, String recipie
     this.text = text;
     this.timestamp = timestamp;
     this.recipient = recipient;
+    this.imageUrl = null;
+  }
+
+  /** Constructs a new Message written by the user, if they uploaded an image. */
+  public Message(UUID id, String user, String text, long timestamp, String recipient,
```

Comment:
You can actually simplify this a little by reusing constructors:
```
  public Message(UUID id, String user, String text, long timestamp, String recipient) {
    this(id, user, text, timestamp, recipient, null);
  }

  /** Constructs a new Message written by the user, if they uploaded an image. */
  public Message(UUID id, String user, String text, long timestamp, String recipient,
      String imageUrl) {
    this.id = id;
    this.user = user;
    this.text = text;
    this.timestamp = timestamp;
    this.recipient = recipient;
    this.imageUrl = imageUrl;
  }
```

Reply:
Whoa! I didn't think of that. That's super sick. Thanks, Dan!

Reply:
In that case, would it make more sense for the first constructor to call the third constructor directly, instead of going through the second constructor?

Reply:
Yeah, I think that's a good idea!  That way all of the "partial" constructors redirect straight to the "full" version.